src/main/java/net/minecraftforge/gametest/GameTestMain.java

```diff
@@ -0,0 +1,195 @@
+            WorldData worldData = storageAccess.getDataTag(registryOps, datapacks);
+            if (worldData == null) {
+                DedicatedServerProperties dedicatedserverproperties = serverSettings.getProperties();
+                LevelSettings levelsettings = new LevelSettings(dedicatedserverproperties.levelName, dedicatedserverproperties.gamemode, dedicatedserverproperties.hardcore,
+                        dedicatedserverproperties.difficulty, false, new GameRules(), datapacks);
+                WorldGenSettings worldgensettings = dedicatedserverproperties.getWorldGenSettings(registryHolder);
+
+                // Forge: Deserialize the DimensionGeneratorSettings to ensure modded dims are loaded on first server load (see SimpleRegistryCodec#decode). Vanilla behaviour only loads from the
+                // server.properties and deserializes only after the 2nd server load.
+                worldgensettings = WorldGenSettings.CODEC
+                        .encodeStart(RegistryWriteOps.create(NbtOps.INSTANCE, registryHolder), worldgensettings)
+                        .flatMap(nbt -> WorldGenSettings.CODEC.parse(registryOps, nbt))
+                        .getOrThrow(false, errorMsg -> {});
+                worldData = new PrimaryLevelData(levelsettings, worldgensettings, Lifecycle.stable());
+            }
+
+            storageAccess.saveDataTag(registryHolder, worldData);
+            Collection<GameTestBatch> testBatches = GameTestRunner.groupTestsIntoBatches(GameTestRegistry.getAllTestFunctions());
+            BlockPos spawnPos = optionSet.valueOf(spawnPosOpt);
+            var server = MinecraftServer.spin(serverThread -> new GameTestServer(serverThread, storageAccess, packRepository, resources, testBatches, spawnPos, registryHolder));
+            Thread thread = new Thread("Server Shutdown Thread") {
+                @Override
+                public void run() {
+                    // We don't need to halt here as GameTestServer always calls System#exit on both crash and normal exit
+                    LogManager.shutdown(); // we're manually managing the logging shutdown on the server. Make sure we do it here at the end.
+                }
+            };
+            thread.setUncaughtExceptionHandler(new DefaultUncaughtExceptionHandler(LOGGER));
+            Runtime.getRuntime().addShutdownHook(thread);
+        } catch (Exception e) {
+            LOGGER.fatal("Failed to start the minecraft server", e);
+        }
+    }
+
+    private static final class BlockPosValueConverter implements ValueConverter<BlockPos> {
+        @Override
+        public BlockPos convert(String value) {
+            String[] split = value.split(",");
+            return new BlockPos(Double.parseDouble(split[0]), Double.parseDouble(split[1]), Double.parseDouble(split[2]));
+        }
+
+        @Override
+        public Class<BlockPos> valueType() {return BlockPos.class;}
```

Comment:
Is there a reason this method and the valuePattern method are all shoved onto a single line?